File: src/main/java/com/back/domain/post/post/service/PostService.java
package com.back.domain.post.post.service;

import com.back.domain.member.member.entity.Member;
import com.back.domain.post.post.entity.Post;
import com.back.domain.post.post.repository.PostRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class PostService {
    private final PostRepository postRepository;

    public long count() {
        return postRepository.count();
    }

    public Optional<Post> findById(int id) {
        return postRepository.findById(id);
    }

    public void modify(Post post, String title, String content) {
        post.setTitle(title);
        post.setContent(content);
    }

    public Post write(Member author, String title, String content) {
        Post post = new Post(author, title, content);
        postRepository.save(post);

        return post;
    }
}

File: src/main/java/com/back/domain/post/post/service/PostService.kt
package com.back.domain.post.post.service

import com.back.domain.member.member.entity.Member
import com.back.domain.post.post.entity.Post
import com.back.domain.post.post.repository.PostRepository
import org.springframework.stereotype.Service
import java.util.Optional


@Service
class PostService(
    private val postRepository: PostRepository
) {

    fun count() = postRepository.count()

    fun findById(id: Int): Optional<Post> =
        postRepository.findById(id)

    fun modify(post: Post, title: String, content: String) {
        post.title = title
        post.content = content
    }

    fun write(author : Member, title : String, content : String) : Post
            = postRepository.save(Post(author, title, content))
}